src/core/tsi/transport_security.h

```diff
@@ -111,6 +111,9 @@ tsi_result tsi_construct_allocated_string_peer_property(
 tsi_result tsi_construct_string_peer_property_from_cstring(
     const char *name, const char *value, tsi_peer_property *property);
 
+/* Destroy plugin registration funcion. */
+void tsi_register_destroy_plugin(void (*destroy)(void));
```

Comment:
I believe Julien proposed this (all TSI implementations have a shared tsi_init and tsi_destroy, which is declared in transport_security_interface.h). 

Alternatively, as you suggested, for each TSI implementation that needs init and destroy, it has its own gRPC plugin. This is probably cleaner. If so, we can delete tsi_init and tsi_destroy from transport_security_interface.h and transport_security.cc, and directly register plugin with gts_tsi_init and gts_tsi_destroy.